java/ql/src/Security/CWE/CWE-652/XQueryInjectionLib.qll

```diff
@@ -0,0 +1,35 @@
+import java
+import semmle.code.java.dataflow.FlowSources
+
+class XQueryInjectionConfig extends TaintTracking::Configuration {
+  XQueryInjectionConfig() { this = "XQueryInjectionConfig" }
+
+  override predicate isSource(DataFlow::Node source) { source instanceof RemoteFlowSource }
+
+  override predicate isSink(DataFlow::Node sink) { sink instanceof XQueryInjectionSink }
+}
+
+/*Find if the executeQuery method is finally called.*/
+predicate executeQuery(MethodAccess ma) {
```

Comment:
It looks like you want to follow a `prepareExpression` call to an eventual `executeQuery` call? If so, you should override `isAdditionalTaintStep` such that `prepareExpression` conveys taint from its argument to its result, then make `executeQuery` the sink.

Example: the RequestForgery query (https://github.com/github/codeql/blob/main/java/ql/src/experimental/CWE-918/RequestForgery.ql#L26) defines `requestForgeryStep` (https://github.com/github/codeql/blob/main/java/ql/src/experimental/CWE-918/RequestForgery.qll#L9), which says that a variety of methods should conduct taint from some argument to their result or another argument.

This is better than hard-coding the structure in terms of `LocalVariableDeclExpr` or similar because using the data-flow infrastructure this way means the flow from `prepareExpression` to `executeQuery` can cross function boundaries, be indirected via locals, fields, arrays... anything the dataflow library can cope with, just like we already use to get from the initial source to the `prepareExpression` call.